pkgs/top-level/all-packages.nix

```diff
@@ -9242,6 +9242,10 @@ with pkgs;
 
   signal-desktop = callPackage ../applications/networking/instant-messengers/signal-desktop { };
 
+  signal-desktop-wayland = signal-desktop.override {
```

Comment:
``config.wayland`` are undocumented flaks from either nixpkgs.config or ~/.config/nixpkgs/config.nix. This should be some nixos option, right?

Reply:
wouldn't it be better if there was a nixpkgs-wide flag for using ozone, and then making all the wrappers check if the current session is wayland before adding the ozone flags.
e.g.
```sh
if [ -n "$WAYLAND_DISPLAY" ]; then extraFlags="--enable-features=UseOzonePlatform --ozone-platform=wayland"; fi
```

Reply:
> wouldn't it be better if there was a nixpkgs-wide flag for using ozone, and then making all the wrappers check if the current session is wayland before adding the ozone flags.
e.g.

The reason why you might not want to do that is due to the fact that some Electron applications behave poorly under Wayland or straight up don't work properly (Brave comes to mind: #133331). A user might want to run the ones which work fine under Wayland but not the problematic ones.

Reply:
Well it would be up to the package to be smart about it, no?

Reply:
Actually maybe even only support a NIXOS_OZONE environment variable so that the builds are the same for everyone.

So check NIXOS_OZONE and WAYLAND_DISPLAY and then add the flags if the package behaves ok

Reply:
> A user might want to run the ones which work fine under Wayland but not the problematic ones.

What would happen if I would open brav now under wayland without any extra flags?

Reply:
I think a global solution would be much nicer and easier to maintain rather than creating wayland variants for almost all electron applications.

Reply:
> What would happen if I would open brav now under wayland without any extra flags?

It would open under XWayland.

Reply:
So can we agree on a global solution? I think it's a quick win...

Reply:
I think we can use this almost everywhere:

```bash
${NIXOS_OZONE:+${WAYLAND_DISPLAY:+ --enable-features=UseOzonePlatform --ozone-platform=wayland}}
```

This would add the flags only under wayland and only if NIXOS_OZONE is set to some value. No feature flags needed and thus no extra compiling.